proc.c

```diff
@@ -3102,9 +3125,10 @@ method_inspect(VALUE method)
     if (data->me->def->type == VM_METHOD_TYPE_ALIAS) {
         defined_class = data->me->def->body.alias.original_me->owner;
     }
-    else {
-        defined_class = method_entry_defined_class(data->me);
+    if (data->me->def->type == VM_METHOD_TYPE_ZSUPER) {
+        method = zsuper_ref_method(method);
     }
+    defined_class = method_defined_class(method);
```

Comment:
I think there is an unintentional change here. As is, the `defined_class` assignment in the `ALIAS` branch above is always dead. It looks like this is part is why there is a change to the inspect test that says `shallow_inspect`. The test doesn't seem related to the change as it doesn't make any `ZSUPER` methods so it could stay the same.

Reply:
Good catch. Indeed, it originally comes from https://github.com/jeremyevans/ruby/commit/c95e7e5329140f640b6497905485761f3336d967#diff-b2553d23e6b1fe76e20608d06c25f6acca06279100f1a9c24febcd79a82fac3cR3059 and I remember wondering if that should be an `else if` but then forgot about it.

Reply:
Done.